tiledb/api/src/array/mod.rs

```diff
@@ -34,6 +37,42 @@ impl Mode {
     }
 }
 
+impl TryFrom<ffi::tiledb_query_type_t> for Mode {
+    type Error = crate::error::Error;
+
+    fn try_from(value: ffi::tiledb_datatype_t) -> TileDBResult<Self> {
+        Ok(match value {
+            ffi::tiledb_query_type_t_TILEDB_READ => Mode::Read,
+            ffi::tiledb_query_type_t_TILEDB_WRITE => Mode::Write,
+            ffi::tiledb_query_type_t_TILEDB_DELETE => Mode::Delete,
+            ffi::tiledb_query_type_t_TILEDB_UPDATE => Mode::Update,
+            _ => {
+                return Err(crate::error::Error::ModeType(
+                    ModeErrorKind::InvalidDiscriminant(value as u64),
+                ))
+            }
+        })
+    }
+}
+
+impl Debug for Mode {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        <Self as Display>::fmt(self, f)
+    }
+}
+
+impl Display for Mode {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        let mode_str = match *self {
+            Mode::Delete => String::from("Mode::Delete"),
```

Comment:
The compiler will generate this for you if you do `#[derive(Debug)]` and then defer to the debug format here (for a simple enum there's not really a meaningful difference)

Reply:
I see you resolved this, have you pushed changes to update it? Or did you want to keep it the way it is?
Either is fine with me. Though I'm pretty sure this is exactly what `#[derive(Debug)]` will give you.